### PR TITLE
Dashboard This should be the correct one!

### PR DESCRIPTION
**Pull request by Oli26**

### CODE REVIEW

File: zeeguu_api/api/teacher_dashboard.py [hunk 1]
```diff
@@ -224,8 +223,8 @@ def check_inv_code(invite_code):
     '''
     c = Cohort.query.filter_by(inv_code=invite_code).first()
     if c is None:
-        return jsonify(1)
-    return jsonify(0)
+        return "Taken"
```
Comment (mircealungu) [inline zeeguu_api/api/teacher_dashboard.py]: A slightly more informative error wouldn't hurt ;)

**Comment by mircealungu** [comment]
Fair enough.

M.

File: zeeguu_api/api/teacher_dashboard.py [hunk 2]
```diff
@@ -215,17 +214,17 @@ def _link_teacher_cohort(user_id, cohort_id):
...
+def inv_code_exists(invite_code):
     '''
         Checks if the inputted invite code is already in use.
 
     '''
     c = Cohort.query.filter_by(inv_code=invite_code).first()
     if c is None:
-        return jsonify(1)
-    return jsonify(0)
+        return "False"
```
Comment (mircealungu) [inline zeeguu_api/api/teacher_dashboard.py]: False -- is not the best answer, i'm sorry :) 
think of the implications: 

```
Q: /invite_code_exists?
A: False

```
I guess /validate_invite_code 

would be ok to have False as a result.

File: zeeguu_api/api/teacher_dashboard.py [hunk 3]
```diff
@@ -31,7 +31,7 @@ def has_permission_for_cohort(cohort_id):
 
 @api.route("/has_permission_for_cohort/<id>", methods=["GET"])
 @with_session
-def has_permission_for_cohorts(id):
+def has_permission_for_cohort_public(id):
```
Comment (mircealungu) [inline zeeguu_api/api/teacher_dashboard.py]: what does public mean here?

Comment (Oli26) [reply zeeguu_api/api/teacher_dashboard.py]: There is already a function has_permissions_for_cohort that is private and used inside other functions. Whereas the routed has_permissions_for_cohort is mainly used on the dashboard to identify if it should attempt to load something from the API.

**Comment by mircealungu** [comment]
Oh!
But then can't you use the underscore in front of the private method names
python convention?

E.g. _fibonacci vs fibonacci

M.

**Pull request merged**